spec/integration/system_pager_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+RSpec.describe TTY::Pager::SystemPager do
+  around :each do |example|
+    Dir.mktmpdir do |dir|
+      FileUtils.cd(dir) do
+        example.run
+      end
+    end
+  end
+
+  it "paginates some text" do
+    output_path = Pathname.new('output.txt')
+
+    described_class.page("Some text", command: "cat > #{output_path}")
+    expect(output_path.read).to eq "Some text"
+  end
+
+  it "allows pagination to happen asynchronously" do
+    output_path = Pathname.new('output.txt')
+    File.open('external_pager.rb', 'w') do |script|
+      script.write(<<~RUBY)
```

Comment:
The 'squiggly' operator doesn't work on older Rubies - sorry! An unindent helper in `spec_helper.rb` could be added for now.

Reply:
Actually, this was solved nicely by pulling the code for these in their own files :)